tests/runtime/test_service_factory.py

```diff
@@ -213,6 +216,91 @@ def test_get_and_filter_modules_respects_included_task_types_and_excluded_module
         assert "ListModule" not in str(clean_modules)
 
 
+def test_assert_compatible_does_not_raise_if_not_enabled():
+    with temp_config(
+        {
+            "runtime": {
+                "service_generation": {
+                    "backwards_compatibility": {
+                        "enabled": False,
+                    }
+                },
+            }
+        },
+        "merge",
+    ):
+        ServicePackageFactory.get_service_package(
+            ServicePackageFactory.ServiceType.INFERENCE
```

Comment:
nit: it looks like this code is implemented to be used for both `TRAINING` and `INFERENCE` so we probably want to either:
- Add tests for training, or
- Only implement it for inference

Reply:
Implemented to only work against `INFERENCE`